examples_utils/notebook_logging/gc_logger.py

```diff
@@ -153,37 +173,65 @@ def __update_payload(cls, output: str, name: str) -> str:
             cls._PAYLOAD[name] = empty_output_type()
 
     @classmethod
-    def __get_notebook_metadata(cls):
-        """Get notebook metadata."""
+    def __store_initial_cell_states(cls):
+        """store the initial state of all cells in notebook."""
 
-        try:
-            while True:
-                if cls.LOG_STATE == "DISABLED":
-                    return
+        if cls.LOG_STATE == "DISABLED":
+            return
 
-                try:
-                    notebook_path = str(ipynbname.path())
-                except:
-                    notebook_path = "failed-to-get-nb-path"
+        try:
+            with open(ipynbname.path()) as notebook:
+                initial_state = nbformat.read(notebook, nbformat.NO_CONVERT)
 
-                # Encode and hash
-                notebook_id = os.getenv("PAPERSPACE_NOTEBOOK_ID")
-                salted_id = notebook_id + datetime.now().strftime("%Y-%m-%d")
-                anonymised_notebook_id = base64.urlsafe_b64encode(
-                    hashlib.md5(salted_id.encode("utf-8")).digest()
-                ).decode("ascii")[:16]
+            # Get list of all code cells
+            for cell in initial_state["cells"]:
+                if cell["cell_type"] == "code":
+                    cls._CODE_CELLS.append(cell["source"])
 
-                notebook_metadata = {
-                    "notebook_path": notebook_path,
-                    "notebook_repo_id": os.getenv("PAPERSPACE_NOTEBOOK_REPO_ID"),
-                    "notebook_id": anonymised_notebook_id,
-                    "cluster_id": os.getenv("PAPERSPACE_CLUSTER_ID"),
-                }
+        except:
+            pass
 
-                for key, val in notebook_metadata.items():
-                    cls.__update_payload(val, key)
+    @classmethod
+    def __manual_termination_polling(cls):
+        """Report if exeuction termination event was some kill signal."""
 
+        try:
+            while True:
                 time.sleep(cls._POLLING_SECONDS)
+        except:
+            cls.__update_payload(1, "manual_cell_termination_event")
+
+    @classmethod
+    def __get_notebook_metadata(cls):
+        """Get notebook metadata."""
+
+        if cls.LOG_STATE == "DISABLED":
+            return
+
+        try:
+            try:
+                notebook_path = ipynbname.path()
+            except:
+                notebook_path = "failed-to-get-nb-path"
+
+            # Encode and hash
+            notebook_id = os.getenv("PAPERSPACE_NOTEBOOK_ID")
+            salted_id = notebook_id + datetime.now().strftime("%Y-%m-%d")
+            anonymised_notebook_id = base64.urlsafe_b64encode(hashlib.md5(salted_id.encode("utf-8")).digest()).decode(
+                "ascii"
+            )[:16]
+
+            notebook_metadata = {
+                "notebook_path": str(notebook_path),
+                "notebook_name": str(notebook_path.stem),
```

Comment:
This will fail if your call to `ipynbname.path` failed

Reply:
Thanks, fixed in the latest commit